compiler/ETA/Utils/JAR.hs

```diff
@@ -282,9 +287,19 @@ mergeClassesAndJars jarLocation compress fileAndContents jarSelectors = do
   p <- makeAbsoluteFilePath jarLocation
   copyFile copy p
   withArchive p $ do
+    existingEntries <- getEntries
+    let invalidEntries = filterWithKey (\k _ -> invalidEntrySelector k)  existingEntries
+    mapM_ deleteEntry (keys invalidEntries)
```

Comment:
This line can have unintended consequences for the user. Was it necessary to fix the problem you faced?